packages/chips/ChipSet.js

```diff
@@ -29,27 +29,35 @@ import ChipCheckmark from './ChipCheckmark';
 
 export default class ChipSet extends Component {
   checkmarkWidth_ = 0;
-  foundation_ = null;
-  state = {
-    selectedChipIds: new Set(this.props.selectedChipIds),
-  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedChipIds: props.selectedChipIds,
+      foundation: null,
+      hasInitialized: false,
+    };
+  }
 
   componentDidMount() {
-    this.foundation_ = new MDCChipSetFoundation(this.adapter);
-    this.foundation_.init();
-    this.updateChipSelection();
+    const foundation = new MDCChipSetFoundation(this.adapter);
+    this.setState({foundation});
+    foundation.init();
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.props.selectedChipIds !== prevProps.selectedChipIds) {
-      const selectedChipIds = new Set(this.props.selectedChipIds);
+  componentDidUpdate(prevProps, prevState) {
+    const {selectedChipIds} = this.props;
+
+    if (this.state.foundation !== prevState.foundation) {
+      this.initChipSelection();
+    }
+    if (selectedChipIds !== prevProps.selectedChipIds) {
       this.setState({selectedChipIds});
```

Comment:
```suggestion
      this.updateChipSelection(selectedChipIds);
```

Reply:
Here we need to update the foundation with the new `props.selectedChipIds` in order to handle single selection logic for choice chips.

and since we're going to update the foundation anyways, we don't need to hold the `selectedChipIds` in state anymore. We can just use `foundation.getSelectedChipIds()` to determine if a chip is selected or not.